File: Intermediate/16.Find the longest common prefix in an array of strings/Solution.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

string longestCommonPrefix(vector<string>& strs) {
    if (strs.empty()) return "";

    string prefix = strs[0];
    for (int i = 1; i < strs.size(); i++) {
        int j = 0;
        while (j < strs[i].size() && j < prefix.size() && strs[i][j] == prefix[j]) {
            j++;
        }
        prefix = prefix.substr(0, j);
    }
    return prefix;
}

int main() {
    int n;
    cout << "Enter the number of strings: ";
    cin >> n;

    vector<string> strs(n);
    cout << "Enter the strings:\n";
    for (int i = 0; i < n; i++) {
        cin >> strs[i];
    }

    string result = longestCommonPrefix(strs);
    if (!result.empty()) {
        cout << "Longest Common Prefix: " << result << endl;
    } else {
        cout << "No common prefix." << endl;
    }

    return 0;
}

File: Intermediate/16.Find the longest common prefix in an array of strings/Solution.py
def longest_common_prefix(strs):
    if not strs:
        return ""

    prefix = strs[0]
    for i in range(1, len(strs)):
        j = 0
        while j < len(strs[i]) and j < len(prefix) and strs[i][j] == prefix[j]:
            j += 1
        prefix = prefix[:j]

    return prefix

if __name__ == "__main__":
    n = int(input("Enter the number of strings: "))
    strs = []
    print("Enter the strings:")
    for _ in range(n):
        strs.append(input())

    result = longest_common_prefix(strs)
    if result:
        print("Longest Common Prefix:", result)
    else:
        print("No common prefix.")
